airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/AirbyteApplication.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.lib;
+
+/**
+ * Interface representing an Airbyte Application to collect metrics for. This interface is present
+ * as Java doesn't support enum inheritance as of Java 17. We use a shared interface so this
+ * interface can be used in the {@link AirbyteMetricsRegistry} enum.
+ */
+public interface AirbyteApplication {
```

Comment:
This name doesn't help me understand what is. I think this is really `MetricsEmittingApp`. You can maybe find a name that rolls of the tongue better than that, but I don't think having the word `Airbyte` in the name tells us anything and if anything a bit misleading.

Reply:
good point - the name works!